indico/modules/events/contributions/controllers/management.py

```diff
@@ -200,14 +201,16 @@ def _process(self):
         form = contrib_form_class(obj=FormDefaults(self.contrib, start_date=self.contrib.start_dt,
                                                    **custom_field_values),
                                   event=self.event_new, contrib=self.contrib, session_block=parent_session_block)
+        non_users_before = find_unregistered_users(self.contrib.speakers)
         if form.validate_on_submit():
             with track_time_changes():
                 update_contribution(self.contrib, *_get_field_values(form.data))
+            flash_if_unregistered(non_users_before, self.contrib.speakers, self.event_new)
             flash(_("Contribution '{}' successfully updated").format(self.contrib.title), 'success')
             tpl_components = self.list_generator.render_list(self.contrib)
             if tpl_components['hide_contrib']:
                 self.list_generator.flash_info_message(self.contrib)
-            return jsonify_data(**tpl_components)
+            return jsonify_data(flash='flash' in request.args, **tpl_components)
```

Comment:
I'd rather do `flash=(request.args.get('flash') == '1')` - even if we don't do it, I always find it weird that `flash=0` would be the exact same as `flash=1` in the query string.